src/convertToObject.js

```diff
@@ -1,18 +1,22 @@
 'use strict';
 
 /**
- * Implement convertToObject function:
- *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
- * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
- *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString.split(';')
+    .filter(style => style.trim().length > 0)
+    .map(style => style.trim())
+    .reduce((styles, style) => {
+      const nameOfStyle = style.split(':')[0].trim();
+      const valueOfStyle = style.split(':')[1].trim();
```

Comment:
Why do you need to split the style twice? Better destructurize it at first call.